tests/test_management_api.py

```diff
@@ -99,6 +101,93 @@ def test_get_policy_api(self):
         self.assertTrue(e.has_grouping_policy(["alice", "data2_admin"]))
         self.assertFalse(e.has_grouping_policy(["bob", "data2_admin"]))
 
+    def test_get_policy_matching_function(self):
+        e = self.get_enforcer(
+            get_examples("rbac_with_domain_and_policy_pattern_model.conf"),
+            get_examples("rbac_with_domain_and_policy_pattern_policy.csv"),
+        )
+        e.get_role_manager().add_domain_matching_func(casbin.util.key_match2_func)
+
+        self.assertEqual(
+            e.get_policy(),
+            [
+                ["admin", "domain.*", "data1", "read"],
+                ["user", "domain.1", "data2", "read"],
+                ["user", "domain.1", "data2", "write"],
+            ],
+        )
+
+        km2_fn = casbin.util.key_match2_func
+        self.assertEqual(
+            e.get_filtered_grouping_policy(2, partial(km2_fn, "domain.3")),
+            [["alice", "user", "*"], ["bob", "admin", "domain.3"]],
+        )
+
+        self.assertEqual(
+            e.get_filtered_grouping_policy(2, partial(km2_fn, "domain.1")),
+            [["alice", "user", "*"]],
+        )
+
+        # first p record matches to domain.3
+        self.assertEqual(
+            e.get_filtered_policy(1, partial(km2_fn, "domain.3")),
+            [["admin", "domain.*", "data1", "read"]],
+        )
+
+        # first and second p record should be matched to (.., domain.1, read)
+        self.assertEqual(
+            e.get_filtered_policy(1, partial(km2_fn, "domain.1"), "", "read"),
+            [
+                ["admin", "domain.*", "data1", "read"],
+                ["user", "domain.1", "data2", "read"],
+            ],
+        )
+
+    def test_get_policy_multiple_matching_functions(self):
+        e = self.get_enforcer(
+            get_examples("rbac_with_domain_and_policy_pattern_model.conf"),
+            get_examples("rbac_with_domain_and_policy_pattern_policy.csv"),
+        )
+        e.get_role_manager().add_domain_matching_func(casbin.util.key_match2_func)
```

Comment:
This test case works without adding a domain matching function.

Reply:
Sorry, the idea was to be on a state where you want to matcher(thus registered with km2_fn) be consistent with  your policy API. But of course they are independent.